src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -1172,6 +1159,46 @@ internal override void CreateAsync()
             SendOneItem();
         }
 
+        private void StartOutputAndErrorReaderThreads(Process serverProcess)
+        {
+            ParameterizedThreadStart func = arg => {
```

Comment:
Will this function be allocated for each transport created?  Would it make more sense to create a class method that takes reader argument or enum that defines data type?  I would prefer to avoid tuples if possible.

Reply:
This method will be made an instance method of the `OutOfProcTransportManager` by the compiler, but I can make it a separate private class method.

Reply:
Add `ProcessOutputData` and `ProcessErrorData` methods, and both take a `StreamReader` only.